engine/src/hc.cpp

```diff
@@ -1998,10 +1998,16 @@ IO_stat MCHcstak::read(IO_handle stream)
 				while (tsize--)
 				{
 					if (IO_read_uint1(&byte, stream) != IO_NORMAL)
+					{
+						delete[] tbuffer;
 						return IO_ERROR;
+					}
 					while ((byte == '\n' || byte == '\r') && tsize--)
 						if (IO_read_uint1(&byte, stream) != IO_NORMAL)
+						{
+							delete[] tbuffer;
```

Comment:
@peter-b after looking at this I'd rather avoid touching this code other than to fix the leak as I've done. I don't think it's worth the time or risk of injecting a bug into HC stack import. I did silence a few warnings that were close by ;-)